### PR TITLE
fix bug where leading zeros on MPInts are not stripped

### PR DESCRIPTION
**Pull request by agrinman**

### CODE REVIEW

File: PGPFormat/PublicKeyData.swift [hunk 1]
```diff
@@ -60,17 +84,57 @@ public struct Ed25519PublicKey:PublicKeyData {
...
+        let mpint = try MPInt(mpintData: Data(bytes: bytes[start ..< bytes.count]))
+        
+        guard mpint.data.bytes.first == Constants.prefixByte else {
+            throw ParsingError.missingECCPrefixByte
+        }
+        
+        self.mpint = mpint
+    }
+    
+    public func keyData() throws -> Data {
```
Comment (kcking) [inline PGPFormat/PublicKeyData.swift]: we could also store the key data without the prefix byte and prepend it on serialization so that this method doesnt need to throw

File: PGPFormat/Util.swift [hunk 2]
```diff
@@ -19,6 +19,60 @@ public enum DataError : Error {
...
 
+public struct MPInt {
+    
+    public var data:Data
+    
+    /**
+        Initialize an MPInt with integer bytes
+        remove any leading zero bytes
+     */
+    public init(integerData:Data) {
```
Comment (kcking) [inline PGPFormat/Util.swift]: does this work on all `0`s?

File: PGPFormat/Util.swift [hunk 3]
```diff
@@ -80,11 +134,21 @@ public extension Data {
...
+        var byteIndex = 0
+        for byte in dataBytes {
+            guard Int(byte) == 0 else {
+                break
+            }
+            
+            byteIndex += 1
+        }
+        
+        let firstByteBits = Int(dataBytes[byteIndex]).numBits
```
Comment (kcking) [inline PGPFormat/Util.swift]: does this work on all `0`s?

File: PGPFormat/Util.swift [hunk 4]
```diff
@@ -80,11 +134,21 @@ public extension Data {
...
+        let remaindingBytesBits = (count - byteIndex - 1)*8
```
Comment (kcking) [inline PGPFormat/Util.swift]: typo: `remaining`

File: PGPFormat/Util.swift [hunk 5]
```diff
@@ -227,6 +312,12 @@ public extension UInt32 {
     }
 }
 
+public extension Int {
+    func twoByteBigEndianBytes() -> [UInt8] {
+        return [UInt8((self >> 8) % 256), UInt8((self) % 256)]
```
Comment (kcking) [inline PGPFormat/Util.swift]: I think the `% 256` is unnecessary here

File: PGPFormat/PublicKeyData.swift [hunk 6]
```diff
@@ -60,17 +84,57 @@ public struct Ed25519PublicKey:PublicKeyData {
...
+        let bytes = mpintData.bytes
+        
+        var start = 0
+        guard Int(bytes[start]) == Constants.curveOID.count else {
+            throw ParsingError.badECCCurveOIDLength(bytes[start])
+        }
+        
+        start += 1
+        
+        let curveOID = [UInt8](bytes[start ..< start + 9])
```
Comment (agrinman) [inline PGPFormat/PublicKeyData.swift]: 👍

File: PGPFormat/PublicKeyData.swift [hunk 7]
```diff
@@ -60,17 +84,57 @@ public struct Ed25519PublicKey:PublicKeyData {
...
+        guard curveOID == Constants.curveOID else {
+            throw ParsingError.unsupportedECCCurveOID(Data(bytes: curveOID))
+        }
+        
+        start += 9
```
Comment (agrinman) [inline PGPFormat/PublicKeyData.swift]: 👍 +1

File: PGPFormat/PublicKeyData.swift (hunk 1 above)
Comment (agrinman) [inline PGPFormat/PublicKeyData.swift]: 👍

File: PGPFormat/Util.swift (hunk 2 above)
Comment (agrinman) [inline PGPFormat/Util.swift]: yes, added a test case for it though

File: PGPFormat/Util.swift (hunk 3 above)
Comment (agrinman) [inline PGPFormat/Util.swift]: 👍 , edge case, added a test for it.

File: PGPFormat/Util.swift (hunk 4 above)
Comment (agrinman) [inline PGPFormat/Util.swift]: 👍

File: PGPFormat/Util.swift (hunk 5 above)
Comment (agrinman) [inline PGPFormat/Util.swift]: it is
